### PR TITLE
DM-1089 :Creating a new deal with ABA format to verify that template type field is displaying

### PR DESCRIPTION
**Pull request by Mkawad**
- Creating a new Bidder via API
- Creating a new Deal-group via API
- Creating a new deal with ABA format via UI
- Verify that template type field is displaying via UI

### CODE REVIEW

File: cypress/integration/deal_template_type.js [hunk 1]
```diff
@@ -0,0 +1,226 @@
+/* eslint-disable linebreak-style */
```
Comment (Frantz-B) [inline cypress/integration/deal_template_type.js:1]: Do not disable the line-break style

File: cypress/integration/deal_template_type.js [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+      cy.get('[data-qa="toggle-tabs--select-PMP"]').click();
+      cy.get('[data-qa="bidder-add-edit--non-secure-url-input"]').type('http://kargo.com');
+      cy.get('[data-qa="bidder-add-edit--secure-url-input"]').type('https://kargo.com');
+      cy.get('[data-qa="bidder-add-edit--send--non-matched"]').click();
+      cy.get('[data-qa="bidder-add-edit--send--iframe-wrapper"]').click();
+      cy.get('[data-qa="bidder-add-edit--send--gzip"]').click();
+      cy.get('[data-qa="bidder-add-edit--send--sync-weight"]').type('0.1');
+      cy.get('[data-qa="bidder-add-edit--save"]').click();
+      cy.get('[data-qa="modal--confirm"]').click();
+    });
```
Comment (Frantz-B) [inline cypress/integration/deal_template_type.js]: If main focus is to Verify ABA format, then creating bidders should be done via API for speed.  There is already a UI test case that creates Bidders

File: cypress/integration/deal_template_type.js [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+      // Validating Bidder was updated in UI
+      cy.get('[class="u-ellipsis"]').should('contain', bidderUpdatedName);
+      cy.get('section.u-grid.layout--details.u-grid-gap-24 aside div div:nth-child(2) p3').should('contain', 'Open and PMP');
+      cy.get('section.u-grid.layout--details.u-grid-gap-24 aside div div:nth-child(3) p3').should('contain', '0.9');
+      cy.get('section.u-grid.layout--details.u-grid-gap-24 aside div div:nth-child(4) p3:nth-child(2)').should('contain', 'Use direct-like cookie sync');
+      cy.get('section.u-grid.layout--details.u-grid-gap-24 aside div div:nth-child(4) p3:nth-child(3)').should('contain', 'Send video traffic');
+      cy.get('section.u-grid.layout--details.u-grid-gap-24 aside div div:nth-child(4) p3:nth-child(4)').should('contain', 'Send Native Traffic');
+      cy.get('section.u-grid.layout--details.u-grid-gap-24 aside div div:nth-child(5) p3').should('contain', 'http://google.com');
+      cy.get('section.u-grid.layout--details.u-grid-gap-24 aside div div:nth-child(6) p3').should('contain', 'https://google.com');
+    });
```
Comment (Frantz-B) [inline cypress/integration/deal_template_type.js]: If main focus is to Verify ABA format, then there is no need to Edit and Validate a Bidder

File: cypress/integration/deal_template_type.js [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+      cy.get('[data-qa="deal-group-dashboard--create"]').click();
+      cy.get('[data-qa="deal-group-form--name"]').focus().type(dealGroupName);
+      cy.get('[data-qa="deal-group-form--sf-id"]').focus().type(salesforceId);
+      cy.get('[data-qa="deal-group-form--target-spend"]').focus().type(targetSpend);
+      cy.get('[data-qa="deal-group-form--closed-loop"]').click().then(() => {
+        cy.get('[data-qa="deal-group-form--closed-loop--1"]').click();
+      });
+      cy.wait(2000);
+
+
```
Comment (Frantz-B) [inline cypress/integration/deal_template_type.js]: Only space is necessary

File: cypress/integration/deal_template_type.js [hunk 5]
```diff
@@ -0,0 +1,226 @@
...
+          .first()
+          .children('a')
+          .click();
+      });
+
+
+      // KPI Section
+      cy.get('[data-qa="deal-group-form--optimizable-kpi"]').click().then(() => {
+        cy.get('[data-qa="deal-group-form--optimizable-kpi--0"]').click();
+      });
```
Comment (Frantz-B) [inline cypress/integration/deal_template_type.js]: If main focus is to Verify ABA format, then creating Deal-Group should be done via API for speed.  There is already a UI test case that creates Deal-Group

File: cypress/integration/deal_template_type.js [hunk 6]
```diff
@@ -0,0 +1,226 @@
...
+
+      // Buyers Section - select bidder
+      cy.get('[data-qa="deal-form--select-bidder"]').focus().clear().type('A')
+        .then(() => {
+          cy.get('[data-qa="deal-form--select-bidder"]').parent('.input-wrapper--typeahead').find('.dropdown-menu')
+            .children('li')
+            .first()
+            .children('a')
+            .click();
+        });
```
Comment (Frantz-B) [inline cypress/integration/deal_template_type.js]: This has to be updated to reflect new UI changes from DM-905

File: cypress/integration/deal_template_type.js [hunk 7]
```diff
@@ -0,0 +1,226 @@
...
+      cy.location().then((currentLocation) => {
+        const urlPathName = currentLocation.pathname;
+        dealID = urlPathName.split('/').pop(); // Grabbing Deal ID from URL
+      });
+    });
+
+    it('Edit Deal', () => {
+      cy.visit(`/deal-dashboard/deals/${dealID}`);
+      cy.get('[class="button button--edit"]').click({ force: true }).then(() => { // Requires 'force: true' for hidden element
+        cy.url().should('include', 'deal-dashboard/deals/edit/');
```
Comment (Frantz-B) [inline cypress/integration/deal_template_type.js:182]: Pls add a validation/assertion section to cross check all the values that were entered to create the deal

File: cypress/integration/deal_template_type.js [hunk 8]
```diff
@@ -0,0 +1,226 @@
...
+      cy.get('input[placeholder="Add ABA Template"]').click().type('A').then(() => {
+        cy.get('typeahead-container ul li:nth-child(2)')
+          .click();
+      });
+
+      // Submitting Deal Group info and Validating
+      cy.get('[data-qa="deal-add-edit--submit"]').click();
+      cy.get('[data-qa="modal--confirm"]').click();
+      cy.get('[data-qa="deal-detail--title"]').should('contain', `${dealName}_edited`);
+    });
```
Comment (Frantz-B) [inline cypress/integration/deal_template_type.js:259]: There should be a validation/assertion section that cross checks all the new values that were updated

**Review by Frantz-B — CHANGES_REQUESTED** [review]
I also recommend one way of achieving the ABA format test, is to first create the Deal with a different format, like how you did with your ticket DM-1090 (selecting bottom banner).  Then after that, when you go to edit the deal, you then select the ABA format.

File: cypress/integration/deal_template_type.js [hunk 9]
```diff
@@ -0,0 +1,217 @@
...
+          .then((resp) => {
+            bidderId = resp.body.id;
+
+            assert.equal(resp.status, 200, ['response status value ']);
+            assert.isAbove(resp.body.id, 0, ['id is greater than 0 ']);
+          });
+      });
+    });
+    it('pushing the created bidder to ad-server', () => {
+      cy.fixture('Bidder-to-adServer.json').then((bidder) => {
```
Comment (Frantz-B) [inline cypress/integration/deal_template_type.js:56]: why does bidder POST need a km_id?

File: cypress/integration/deal_template_type.js [hunk 10]
```diff
@@ -0,0 +1,217 @@
...
+          method: 'POST',
+          body: reqBody,
+          url: 'https://deal-manager.dev.kargo.com/api/v1/deal/template',
+        });
+        cy.request(requestOptions)
+          .then((resp) => {
+            assert.equal(resp.status, 200, ['response status value ']);
+          });
+      });
+    });
```
Comment (Frantz-B) [inline cypress/integration/deal_template_type.js:243]: What is the purpose of this function?

**Review by Frantz-B — CHANGES_REQUESTED** [review]
I doubt the bidder-to-adServer.json file is need but explain to me why it is.  Other than that and changes I requested, add PR to DM automation repo

File: cypress/integration/deal_template_type.js (hunk 10 above)
Comment (Mkawad) [reply cypress/integration/deal_template_type.js:243]: This function is needed to add a template to the deal

File: cypress/integration/deal_template_type.js (hunk 9 above)
Comment (Mkawad) [reply cypress/integration/deal_template_type.js:56]: To push the bidder to ad-server due to the deal only being pushed with a synced bidder

**Comment by Frantz-B** [comment]
Closing this PR. but we do need aba template

**Pull request closed without merging**